File: internetshoporaclesql/src/main/java/com/project/internetshop/model/Product.java
package com.project.internetshop.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Product {

  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  private Long id;

  @ManyToOne(targetEntity = Category.class)
  private Category category;

  private String name;
  private int price;
  private String imageUrl;
  private boolean isAvailable;
}

File: internet-shop-angular/src/app/model/product.ts
import {Category} from "./category";

export class Product{
  id: number;
  category: Category;
  name: string;
  price: number;
  imageUrl: string;
  available: boolean;


  constructor(category: Category, name: string, price: number, imageUrl: string, available: boolean) {
    this.category = category;
    this.name = name;
    this.price = price;
    this.imageUrl = imageUrl;
    this.available = available;
  }

  setCategory(category: Category): void{
    this.category = category;
  }
}
